.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+on: [push, pull_request]
+name: build
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        ghc: ['8.8.4','8.10.7']
+        os: [ubuntu-latest]
+    name: Haskell GHC ${{ matrix.ghc }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1.2.9
+        id: setup-haskell-build
+        with:
+          ghc-version: ${{ matrix.ghc }} # Resolves to the latest point release of GHC 8.8
+          cabal-version: '3.6.2.0' # Exact version of Cabal
+          enable-stack: true
+          stack-version: '2.7'
+      - name: Install dependencies
+        run: sudo apt-get install -y libxml2-utils libgl1-mesa-dev libglu1-mesa-dev freeglut3-dev libpulse-dev libblas-dev liblapack-dev libasound2-dev
+      - name: Cabal update
+        run: cabal update
+      - name: Freeze Cabal plan
+        run: cabal freeze
+      - name: Cache Cabal build artifacts
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.setup-haskell-build.outputs.cabal-store }}
+            dist-newstyle
+          key: ${{ runner.os }}-cabal-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+            ${{ runner.os }}-cabal-${{ matrix.ghc }}
+      - name: Cabal build dependencies
+        run: |
+          cabal build all --enable-tests --only-dependencies
+      - name: Cabal build all
+        run: |
+          cabal build all --enable-tests
+      - name: Cabal test
+        run: |
+          cabal test all --enable-tests --test-show-details=Always
+      - name: Cache stack build artifacts
```

Comment:
There is no clever way to have a cache key depend on the stack lock file? Would it be a good idea to have that?

Reply:
I believe it already does. Looking at the source of the action [here](https://github.com/freckle/stack-cache-action/blob/main/src/stack-cache.js) it will hash the lock file, the cabal files and the source code. It uses the latest cache available from the more specific to the least specific. So if the source changes but the cabal files remain the same, it uses latest cache available with same cabal files, if cabal changes it uses the latest cache with the same lock file and if the lock file changes it does everything from scratch.

Reply:
Right! That's great.